src/Microsoft.Azure.ServiceBus/ClientEntity.cs

```diff
@@ -98,6 +98,7 @@ internal set
             if (callClose)
             {
                 await this.OnClosingAsync().ConfigureAwait(false);
+                await this.ServiceBusConnection.CloseAsync().ConfigureAwait(false);
```

Comment:
A client might not be always be able to close a connection. Example: connection that it didn't create (doesn't own it) shouldn't not be closed by the client, but by the creator.

Reply:
Given that `OwnsConnection` property is now part of this class, we should move that logic here.
Basically close the connection only if that property is true.

---
In reply to: [196840576](https://github.com/Azure/azure-service-bus-dotnet/pull/506#discussion_r196840576) [](ancestors = 196840576)

Reply:
ok, I will move all logic related to the closing connection here